cmd/occollector/app/collector/processors.go

```diff
@@ -315,12 +317,23 @@ func startProcessor(v *viper.Viper, logger *zap.Logger) (consumer.TraceConsumer,
 			"Found global attributes config",
 			zap.Bool("overwrite", multiProcessorCfg.Global.Attributes.Overwrite),
 			zap.Any("values", multiProcessorCfg.Global.Attributes.Values),
+			zap.Any("key-mapping", multiProcessorCfg.Global.Attributes.KeyReplacements),
 		)
-		tp, _ := addattributesprocessor.NewTraceProcessor(
-			multiconsumer.NewTraceProcessor(traceConsumers),
-			addattributesprocessor.WithAttributes(multiProcessorCfg.Global.Attributes.Values),
-			addattributesprocessor.WithOverwrite(multiProcessorCfg.Global.Attributes.Overwrite),
-		)
+
+		var tp processor.TraceProcessor
+		if len(multiProcessorCfg.Global.Attributes.Values) > 0 {
+			tp, _ = addattributesprocessor.NewTraceProcessor(
+				multiconsumer.NewTraceProcessor(traceConsumers),
+				addattributesprocessor.WithAttributes(multiProcessorCfg.Global.Attributes.Values),
+				addattributesprocessor.WithOverwrite(multiProcessorCfg.Global.Attributes.Overwrite),
+			)
+		}
+		if len(multiProcessorCfg.Global.Attributes.KeyReplacements) > 0 {
+			if tp == nil {
+				tp = multiconsumer.NewTraceProcessor(traceConsumers)
+			}
+			tp, _ = attributekeyprocessor.NewTraceProcessor(tp, multiProcessorCfg.Global.Attributes.KeyReplacements...)
+		}
 		return tp, closeFns
```

Comment:
This will return `tp=nil` if both `Values` and `KeyReplacements` are empty, is that OK?

Reply:
good catch, I had manually tested with `values: {}`, `key-mapping: {}`, and `attributes: {}` but if there is `attributes.overwrite: true` it will cause to return nil and crash. Adding a test to cover this case.